ui/src/app/components/webcomponents/action-step-form.component.ts

```diff
@@ -1356,8 +1347,10 @@ export class ActionStepFormComponent extends BaseComponent implements OnInit {
     for (let i = 0; i < testDataPlaceHolders.length; i++) {
       const content = testDataPlaceHolders[i].innerHTML;
       if ((testDataPlaceHolders[i]?.contentEditable||Boolean(isReplacer)) && ((testDataPlaceHolders[i].getAttribute("data-test-data-type") == null) ||
-        ((testDataPlaceHolders[i].getAttribute("data-test-data-type") != null && (Boolean(isReplacer) ||content == '@| |' || content == '!| |' || content == '*| |'
-          || content == "$| <span class='test_data_place'></span> |" || content == "~| <span class='test_data_place'></span> |" || content == "" || !this.removeHtmlTags(testDataPlaceHolders[i].textContent).trim().length))))) {
+        ((testDataPlaceHolders[i].getAttribute("data-test-data-type") != null && (Boolean(isReplacer) ||
+          contains(content,"/\@\|.*\|/","i") || contains(content,"/\!\|*\|/","i") ||
+          contains(content,"/\*\|.*\|/","i") || contains(content,"/\$\|.*\|/","i") ||
+          contains(content,"/\~\|.*\|/","i")|| contains(content,"/^[a-zA-Z]*$") || !this.removeHtmlTags(testDataPlaceHolders[i].textContent).trim().length))))) {
```

Comment:
why do we need to check for all type of test data pattern? what is the use case here

Reply:
to check wether the testdataplaceholder having any data or not